### ISSUE TITLE
Should we rename PrimitiveAtom to BaseAtom?

### ISSUE THREAD

**Issue by dai-shi** [pmndrs/jotai#238]
Before releasing a new version with #233, we would like to discuss about renaming in types.

pros:
- `BaseAtom` is shorter
- "Primitive" atoms might be confused with primitive values (= non objects)

cons:
- Its' a breaking change
- Docs should be modified accordingly 

It's a good timing of this change, because #233 add some changes in types anyway.

**Comment by Aslemammad** [comment]
I was thinking about it, And my opinion is "YES", I think we have to change it.

**Comment by dai-shi** [comment]
I was thinking about it and found a weird case:

```
"primitive" -> "derived" -> "derived"
```

This is fine, but

```
"base" -> "derived" (A) -> "derived" (B)
```

in this case, (A) can be said the "base" thing of (B).

In both cases, `->` means "derive".

cc @merisbahti

**Comment by dai-shi** [comment]
During the conversation with @JLarky in discord, I noticed I did something wrong.
In #235, I wanted to make a precise type of the return value of `atom()`, and named it `PrimitiveAtom` which is the same as before.
The problem was I confused with types for internal implementation and types for API.
I guess developers doesn't need to care the type `WritableAtom<Value, SetStateAction<Value>>` is implemented with `atom(value)` or derived atoms. There should be no difference when using.

The issue #231 is an issue, but we don't need to change the public API types. That issue should be solved sooner or later, meanwhile I'd add a type with internal one.

Sorry for the mess, now I feel like I should revert `PrimitiveAtom` to be just an alias of `WritableAtom<Value, SetStateAction<Value>>`. We can discuss about renaming it still, but I'm less motivated as there are no breaking changes in types with the revert.
